tests/test_hf_conversion_script.py

```diff
@@ -48,6 +176,183 @@ def get_config(
     return cast(DictConfig, test_cfg)
 
 
+@pytest.mark.world_size(2)
+@pytest.mark.gpu
+@pytest.mark.parametrize('model', ['mpt', 'neo', 'llama2'])
+@pytest.mark.parametrize('fsdp_state_dict_type', ['full', 'sharded'])
+def test_huggingface_conversion_callback(model: str, tmp_path: pathlib.Path,
+                                         fsdp_state_dict_type: str):
+    delete_transformers_cache()
+
+    dist.initialize_dist(get_device('gpu'))
+
+    max_seq_len = 16
+    save_interval_batches = 2
+    huggingface_save_interval_batches = 3
+    device_batch_size = 1
+    dataset_size = 14
+    max_duration_batches = 7
+    precision_str = 'bfloat16'
+    precision = torch.bfloat16
+
+    checkpointer_callback = HuggingFaceCheckpointer(
+        save_folder=os.path.join(tmp_path, 'checkpoints'),
+        save_interval=f'{huggingface_save_interval_batches}ba',
+        precision=precision_str,
+    )
+
+    # get small version of each model
+    model_cfg = None
+    if model == 'mpt':
+        model_cfg = get_config(
+            conf_path='scripts/train/yamls/pretrain/testing.yaml')
+    elif model == 'neo':
+        model_cfg = get_config(
+            conf_path='scripts/train/yamls/pretrain/gpt-neo-125m.yaml')
+        model_cfg['model']['config_overrides']['hidden_size'] = 36
+    elif model == 'llama2':
+        if 'HUGGING_FACE_HUB_TOKEN' not in os.environ:
+            pytest.skip(
+                'The CI cluster does not have access to the Llama models, so skip this test.'
+            )
+        model_cfg = get_config(
+            conf_path='scripts/train/yamls/pretrain/gpt-neo-125m.yaml')
```

Comment:
Can you put a yaml in the test folder that matches what you want to do? We shouldn't rely on other configs in very different folders, and that way you also won't need to modify the config programmatically

Reply:
I decided to just create the config dictionaries in the test and not rely on yamls. I also made a jira to add useful fixtures to foundry so we can stop copying these configs around in different tests.